Select multiple items

icidasset opened the pull request:
Adds:
- [x] Range selection (shift-key + click)
- [x] Individual selection (cmd/meta-key + click)
- [x] Details sidebar for multiple items

Other important changes:
- [x] Integrated the "selected path" into the directory list (ie. directory list now has `selection` next to `items`)

matheus23 commented:
In case you've closed #103 because you wanted a draft PR (just guessing), you can [switch PR between draft and ready for review](https://github.blog/changelog/2020-04-08-convert-pull-request-to-draft/) at any time.

icidasset commented:
@matheus23 That was what I was looking for, thanks 😄

icidasset commented:
@matheus23 Heads up, this changes quite a bit regarding the active "selection".
But basically it boils down to that I replaced `model.selectedPath` and updated `model.directoryList` to the type:
```elm
 Result String { floor : Int, items : List Item, selection : List { index : Int, isFirst : Bool } }
```
